docs/decisions/0003-lightweight-extension-points.rst

```diff
@@ -0,0 +1,51 @@
+Lightweight extension points
+############################
+
+Status
+******
+
+**Draft** *2024-02-27*
+
+Context
+*******
+
+Open Responses are commonly used in education for assessment purposes and the flexibility of ORA has made it a key feature of the Open edX platform. However, as developers, it is tough to accommodate the needs of educators and other stakeholders in the teaching process when they want to enhance the learner's experience with other tools such as plagiarism detection, AI grading, coding graders, and others.
+
+As the code is open, it is always possible to modify it. Still, it might result in a technical maintenance debt, which usually prevents educators from experimenting or installations from upgrading to newer versions once the changes to ORA become important in the teaching process.
+
+Decisions
+*********
+
+As the first step towards making implementing new use cases during the ORA user's lifecycle more flexible, we will introduce two extension points in the ORA codebase. These extension points will be built on top of the `Hooks Extensions Framework`_. The definitions for these extension points will reside in `openedx-filters`_ and `openedx-events`_. These definitions will be imported into the edx-ora2 repository and triggered in two places during the learners' lifecycle, resulting in minimal modifications to the ORA implementation.
```

Comment:
This is very helpful, but I'm not sure I understand what purpose the filter plays here.  Is it to display the result from TurnItIn if it exists to the instructor grading the work?

Reply:
A warning must be displayed before sharing information with external tools so the user knows what's being submitted will be shared with another service. In this case, this is done by implicitly accepting an EULA agreement before sending the information. Now, if the user disagrees with it, it'd be enough not to submit their assessment. We're adding this warning by modifying the ORA template in a plugin:
![image](https://github.com/openedx/edx-ora2/assets/64440265/a2af5d1b-b947-4aa6-b35c-2b9be37924c9)

But this filter could be used for another type of use case where the submission view needs modification or extra functionality; we went for implicit compliance for time constraints, but it could be something even more complex.